apps/server/src/adapters/WebsocketAdapter.ts

```diff
@@ -69,20 +76,37 @@ export class SocketServer implements IAdapter {
           ws.close();
         }
 
-        // TODO: protocol specific stuff should be handled here
-        // eg: rename-client
-        // socket.on('rename-client', (newName) => {
-        //   if (newName) {
-        //     const previousName = this._clientNames[socket.id];
-        //     this._clientNames[socket.id] = newName;
-        //     this.info('CLIENT', `Client ${previousName} renamed to ${newName}`);
-        //   }
-        // });
-
         try {
           const message = JSON.parse(data);
           const { type, payload } = message;
 
+          if (type === 'get-client-name') {
+            ws.send(
+              JSON.stringify({
+                type: 'client-name',
+                payload: clientId,
+              }),
+            );
+            return;
+          }
+
+          if (type === 'set-client-name') {
+            if (payload) {
+              const previousName = clientId;
+              clientId = payload;
+              this.clientIds.delete(previousName);
+              this.clientIds.add(clientId);
```

Comment:
So it's basically this:
```ts
this.clientIds.delete(clientId);
this.clientIds.add(payload); 
```

Kinda confusing, IMO. If these reassignments are simply for renaming, I'd suggest putting these `if` statements into functions so the context can be better understood:
```ts
if (payload) {
  return renameClientId(payload, clientId)
}

// renameClientId
const renameClientId = (newClientId, prevClientId) => { /* function stuff */}
```

We also gain the added benefit of self-documenting code (`renameClientId` vs `set-client-name`). But I understand if you wouldn't want to do that.

Reply:
This is really good input-

I was considering something along those lines, especially for the client-name stuff, which is repeated throughout the code

I wonder whether this version of the code is easier to leverage and maintain since it is more linear. The API calls are completely unrelated and any repeated code is purely coincidental

Reply:
I find smaller contexts easier to understand and repeated code more confusing since you have to be sure the repeated code is exactly the same as everywhere else

Reply:
Interesting. while I would generally agree I wonder on whether this specific example is the wrong context.
I also notice that I tend to do this and get yelled at by sonarqube....

I am inclined to prefer repetition as long as the contexts are isolated.
In this case every if statement runs without care for the remaining context and repetitions are coincidental IMO, this means that the changes would not necessarily replicate across all the usages. I see that there could be a danger for forced abstraction, which I see in many code bases where developers start adding optional flags. Reminds me of a talk I saw some years back from react advance, will try to find it for a future 

Extracting stuff into functions is nice, but tends to make the reader to have to jump a bit around the code, I see it as unnecessary where the code in context is under 5 lines.

I imagine that there is a more elegant pattern to do this, perhaps writing a small middleware that can attach keys to functions as you would do in a route - controller. This could give the best of both worlds, with isolation, linear readability and good semantics. One for a rainy day...

Good discussion

Reply:
> which I see in many code bases where developers start adding optional flags

Guilty as charged. 

While this all makes sense, I'll agree to disagree, lol.